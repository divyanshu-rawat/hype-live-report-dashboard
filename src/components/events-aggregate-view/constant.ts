export const sampleData = [
  {
    displayName: '250v4 Rose Red Kids',
    stock: 500,
    participants: 900,
    sellThrough: 92.2,
    revenue: 130000,
  },
];
