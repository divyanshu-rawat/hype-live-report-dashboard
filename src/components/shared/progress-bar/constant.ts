export const progressBarHeight = {
  md: 16,
  sm: 8,
  xl: 24,
};

export type progressBarSizeType = 'sm' | 'md' | 'xl';
