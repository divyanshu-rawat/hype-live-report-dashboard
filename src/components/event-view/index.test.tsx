import React from 'react';

import { cleanup, render } from '@testing-library/react';

import { EventView } from '.';
import { sampleData } from './constant';

afterEach(cleanup);

test('renders EventView', () => {
    render(<EventView events={sampleData} />);
});

test('title should equal to Sale events', () => {
    const { getByTestId } = render(<EventView events={sampleData} />);
    expect(getByTestId('EventViewTitle')).toHaveTextContent('Sale events');
});

it('should take a EventView snapshot', () => {
    const { asFragment } = render(<EventView events={sampleData} />);
    expect(asFragment()).toMatchSnapshot();
});
