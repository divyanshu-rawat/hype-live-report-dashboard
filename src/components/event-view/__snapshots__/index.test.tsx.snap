// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a EventView snapshot 1`] = `
<DocumentFragment>
  <h2
    class="css-lgmax0"
    data-testid="EventViewTitle"
  >
    Sale events
  </h2>
  <div
    class="css-fzdm0g"
  >
    <div
      class="css-1sr9ro9"
    >
      <h2
        class="css-lr3lix"
        data-testid="EventCardTitle"
      >
        250v4 Rose Red Kids
      </h2>
      <div
        class="css-12xxxyo"
      >
        <div
          class="css-1685clg"
          width="100%"
        >
          <p
            class="css-1ety6vy"
          >
            Sell through
          </p>
          <span
            data-testid="ProgressBarPercentage"
          >
            92.2%
          </span>
        </div>
        <div
          class="css-z2ugs0"
          type="sm"
          width="100%"
        />
      </div>
      <div
        class="css-uf1ume"
      >
        <div
          class="css-1sg2lsz"
        >
          <svg
            aria-label="UnitSold"
            class="css-12bysxo"
            fill="rgba(12, 168, 77, 1)"
            viewBox="0 0 68 68"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M65.249 32.0333L35.8157 2.59999C34.5823 1.36666 32.8823 0.666656 31.1157 0.666656H7.21568C3.54901 0.666656 0.549011 3.66666 0.549011 7.33332V31.2333C0.549011 33 1.24901 34.7 2.51568 35.9333L31.949 65.3667C34.549 67.9667 38.7823 67.9667 41.3824 65.3667L65.2824 41.4667C67.8824 38.8667 67.8824 34.6667 65.249 32.0333ZM36.649 60.6667L7.21568 31.2333V7.33332H31.1157L60.549 36.7667L36.649 60.6667Z"
            />
            <title>
              UnitSold
            </title>
            <path
              d="M15.549 20.6667C18.3104 20.6667 20.549 18.4281 20.549 15.6667C20.549 12.9052 18.3104 10.6667 15.549 10.6667C12.7876 10.6667 10.549 12.9052 10.549 15.6667C10.549 18.4281 12.7876 20.6667 15.549 20.6667Z"
            />
          </svg>
          <p
            class="css-1ety6vy"
          >
            461
          </p>
        </div>
        <div
          class="css-1sg2lsz"
        >
          <svg
            aria-label="PeopleGroup"
            class="css-1gbkm9a"
            fill="rgba(136, 109, 76, 1)"
            viewBox="0 0 81 40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              PeopleGroup
            </title>
            <path
              d="M40.8823 22.5C46.3157 22.5 51.1157 23.8 55.0157 25.5C58.6157 27.1 60.8823 30.7 60.8823 34.6V40H20.8823V34.6333C20.8823 30.7 23.149 27.1 26.749 25.5333C30.649 23.8 35.449 22.5 40.8823 22.5ZM14.2157 23.3333C17.8823 23.3333 20.8823 20.3333 20.8823 16.6667C20.8823 13 17.8823 10 14.2157 10C10.549 10 7.54899 13 7.54899 16.6667C7.54899 20.3333 10.549 23.3333 14.2157 23.3333ZM17.9823 27C16.749 26.8 15.5157 26.6667 14.2157 26.6667C10.9157 26.6667 7.78232 27.3667 4.94899 28.6C2.48232 29.6667 0.882324 32.0667 0.882324 34.7667V40H15.8823V34.6333C15.8823 31.8667 16.649 29.2667 17.9823 27ZM67.549 23.3333C71.2157 23.3333 74.2157 20.3333 74.2157 16.6667C74.2157 13 71.2157 10 67.549 10C63.8823 10 60.8823 13 60.8823 16.6667C60.8823 20.3333 63.8823 23.3333 67.549 23.3333ZM80.8823 34.7667C80.8823 32.0667 79.2823 29.6667 76.8157 28.6C73.9823 27.3667 70.849 26.6667 67.549 26.6667C66.249 26.6667 65.0157 26.8 63.7823 27C65.1157 29.2667 65.8823 31.8667 65.8823 34.6333V40H80.8823V34.7667ZM40.8823 0C46.4157 0 50.8823 4.46667 50.8823 10C50.8823 15.5333 46.4157 20 40.8823 20C35.349 20 30.8823 15.5333 30.8823 10C30.8823 4.46667 35.349 0 40.8823 0Z"
            />
          </svg>
          <p
            class="css-1ety6vy"
          >
            900
          </p>
        </div>
        <div
          class="css-1sg2lsz"
        >
          <svg
            aria-label="Dollar"
            class="css-mo5cu3"
            fill="rgba(33, 150, 243, 1)"
            viewBox="0 0 68 68"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Dollar
            </title>
            <path
              d="M33.8823 0.666656C15.4823 0.666656 0.54895 15.6 0.54895 34C0.54895 52.4 15.4823 67.3333 33.8823 67.3333C52.2823 67.3333 67.2156 52.4 67.2156 34C67.2156 15.6 52.2823 0.666656 33.8823 0.666656ZM33.8823 60.6667C19.1823 60.6667 7.21562 48.7 7.21562 34C7.21562 19.3 19.1823 7.33332 33.8823 7.33332C48.5823 7.33332 60.549 19.3 60.549 34C60.549 48.7 48.5823 60.6667 33.8823 60.6667ZM36.849 31C30.9156 29.0333 28.049 27.8 28.049 24.6667C28.049 21.2667 31.749 20.0333 34.0823 20.0333C38.449 20.0333 40.049 23.3333 40.4156 24.5L45.6823 22.2667C45.1823 20.8 42.949 15.9 36.8156 14.8333V10.6667H30.9823V14.8667C22.3156 16.7333 22.249 24.3667 22.249 24.7333C22.249 32.3 29.749 34.4333 33.4156 35.7667C38.6823 37.6333 41.0156 39.3333 41.0156 42.5333C41.0156 46.3 37.5156 47.9 34.4156 47.9C28.349 47.9 26.6156 41.6667 26.4156 40.9333L20.8823 43.1667C22.9823 50.4667 28.4823 52.4333 30.949 53.0333V57.3333H36.7823V53.2C38.5156 52.9 46.849 51.2333 46.849 42.4667C46.8823 37.8333 44.849 33.7667 36.849 31Z"
            />
          </svg>
          <p
            class="css-1ety6vy"
          >
            130000
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
